### PR TITLE
[1020] Missing details component

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

Adding logic to the `check-details` page which renders a link and text to sections that haven't been started, or are uncompleted.

### Changes proposed in this pull request

- Add new missing details component
- Add custom CSS
- Create logic to make component dynamic
- Add missing details text to locale
- Create view preview

### Guidance to review
- Navigate to this page: https://rtt-review-pr-521.herokuapp.com/trainees/strj7LFNEeZuhddQVpgtm3T2/check-details
- Click the `Start section` link and complete the section without ticking the 'completed' checkbox at the end
- The text and link should have changed on the `/check-details` page
- Repeat for each section
- Check the view previews

https://rtt-review-pr-521.herokuapp.com/view_components/pages/trainees/check_details/view/

#### Dumb components
`BlueInsetTextWithLink::View`
- only get feed with values
- it a wrapper for `GovukComponent::InsetText`
- and displays as is
https://rtt-review-pr-521.herokuapp.com/view_components/blue_inset_text_with_link/view/default

#### Smart components
`Trainees::Sections::View`
- it decides which component to renders
- its a wrapper for any confirmation components as well as `BlueInsetTextWithLink::View` over various statues
- it deals with trainee progress
https://rtt-review-pr-521.herokuapp.com/view_components/trainees/sections/view

### Screenshot
image

### CODE REVIEW

**Comment by avinhurry** [comment]
[[M] Give the review record page component state (or another component)](https://trello.com/c/8bTmKcGX/1020-m-give-the-review-record-page-component-state-or-another-component)

File: app/components/missing_details/style.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+.govuk-inset-text {
+border-left: 5px solid #1d70b8;
+}
+
+.govuk-heading-s {
+  color: #1d70b8;
+  line-height: 0;
+  margin-bottom: 10px
+}
```
Comment (edwardhorsford) [inline app/components/missing_details/style.scss:11]: @avinhurry can we review these? we shouldn't redefine govuk styles. I think we can probably make do with a modifier on the existing inset text component.

**Review by edwardhorsford — CHANGES_REQUESTED** [review]

**Comment by defong** [comment]
heroku issues

**Comment by defong** [comment]
it evolved to #540

**Pull request closed without merging**